examples/lighting-app/linux/main.cpp

```diff
@@ -77,6 +77,41 @@ void ApplicationInit()
 #endif
 }
 
+#if defined(CHIP_IMGUI_ENABLED) && CHIP_IMGUI_ENABLED
+
+class UiAppMainLoopImplementation : public AppMainLoopImplementation
+{
+public:
+    void RunMainLoop() override;
+    void SignalSafeStopMainLoop() override;
+};
+
+void UiAppMainLoopImplementation::RunMainLoop()
+{
+    // Guaranteed to be on the main task (no chip event loop started yet)
+    example::Ui::Init();
+
+    // Platform event loop will be on a separate thread,
+    // while the event UI loop will be on the main thread.
+    chip::DeviceLayer::PlatformMgr().StartEventLoopTask();
+
+    // StopEventLoop will stop the loop below. It is called
+    // from within SignalSafeStopMainLoop below and
+    // UI knows how to stop itself if windows are closed.
+    example::Ui::EventLoop();
+
+    // Stop the chip main loop as well. This is expected to
+    // wait for the task to finish.
+    chip::DeviceLayer::PlatformMgr().StopEventLoopTask();
```

Comment:
Followup: We should still schedule the shutdown event...  `DispatchShutDownAndStopEventLoop` needs to have its first bit called, basically, but no the second.

Reply:
Is this a change that I should do here? I.e. `PlatformMgr().ScheduleWork([](intptr_t) { PlatformMgr().HandleServerShuttingDown(); });` 

Or should I leave it as is and we figure out any better logic that is more signal/thread safe separately? I believe the HandleServerShuttingDown right now only has the ShutDown event send sideffect, this is why I don't actually notice anything wrong in shutdown just by observing app stop.

Reply:
I think we should have an API on Server to do the `PlatformMgr().ScheduleWork([](intptr_t) { PlatformMgr().HandleServerShuttingDown(); });` bit.  But that API should _not_ also stop the event loop.

Reply:
ok, leaving it for followup then.